python/cudf_polars/tests/expressions/test_agg.py

```diff
@@ -74,7 +75,11 @@ def test_agg(df, agg):
 
     # https://github.com/rapidsai/cudf/issues/15852
     check_dtypes = agg not in {"n_unique", "median"}
-    if not check_dtypes and q.collect_schema()["a"] != pl.Float64:
+    if (
+        not check_dtypes
+        and q.collect_schema()["a"] != pl.Float64
+        and DEFAULT_BLOCKSIZE_MODE == "default"
```

Comment:
This is the new condition. The IR graph produced by the multi-partition executor includes some `Cast` nodes that mean we *do* match polars, so we *don't* raise an `AssertionError` in this block (since we've worked around https://github.com/rapidsai/cudf/issues/15852).

Reply:
Question: Should we introduce those casts in the in-memory executor instead (during translation time).

Reply:
I'll plan to make a PR fixing https://github.com/rapidsai/cudf/issues/18615 (by working around #15852). Then everything will be consistent.